### PR TITLE
Implement Infinite Scroll and Update Ingredient Search Page to Utilize Autocomplete Search Form

### PR DESCRIPTION
**Pull request by cathydao95**
## Overview
- Updated API calls to use axios 
- Updated getRecipes to include pagination and limits to control the amount of data retrieved
- Implemented infinite scroll component to enhance user experience when browsing recipes
- Updated Search Ingredients page by removing checklist ingredients and implementing search functionality with autocomplete search. 
- Created additional ingredient router, controller, and table in the DB to handle ingredient related requests. 

https://github.com/cathydao95/recipe-ready/assets/79618165/6df24bc5-772a-4d74-8b25-12006aaef927

### CODE REVIEW

File: client/src/components/ResultsLayout/ResultsLayout.jsx [hunk 1]
```diff
@@ -10,13 +9,13 @@ const ResultsLayout = ({ recipes, title, page }) => {
...
-        <h2 className={styles.title}>{title}</h2>
+        <h1 className={styles.title}>{title}</h1>
         {!recipes || recipes.length === 0 ? (
           <EmptyPageContent page={page} />
         ) : (
           <div className={styles.recipesContainer}>
-            {recipes.map((recipe) => (
-              <RecipeCard key={recipe.id} recipe={recipe} page={page} />
+            {recipes.map((recipe, index) => (
+              <RecipeCard key={index} recipe={recipe} page={page} />
```
Comment (rkomorn) [inline client/src/components/ResultsLayout/ResultsLayout.jsx:18]: Why did you move away from the recipe ID as key?

File: client/src/pages/Results/Results.jsx [hunk 2]
```diff
@@ -1,36 +1,54 @@
 import { useEffect, useState } from "react";
-import { Loading, ResultsLayout } from "../../components";
+import { ResultsLayout } from "../../components";
 import { useAppContext } from "../../context/appContext";
 import { useLocation } from "react-router-dom";
+import InfiniteScroll from "react-infinite-scroll-component";
```
Comment (rkomorn) [inline client/src/pages/Results/Results.jsx:5]: 🥳

File: client/src/pages/Search/SearchByIngredients.jsx [hunk 3]
```diff
@@ -1,37 +1,138 @@
...
     navigate("/results", {
       state: { ingredients: selectedIngredients },
     });
   };
 
+  // Function to fetch autocomplete suggestions
+  const getAutoCompletedIngredients = async (input) => {
+    try {
+      let response = await axios.get(
+        `/api/v1/ingredients/autocomplete?input=${input}`
```
Comment (rkomorn) [inline client/src/pages/Search/SearchByIngredients.jsx:71]: niiiice

**Review by rkomorn — APPROVED** [review]

File: client/src/components/ResultsLayout/ResultsLayout.jsx (hunk 1 above)
Comment (cathydao95) [reply client/src/components/ResultsLayout/ResultsLayout.jsx:18]: Oh good catch! I had intended to do this for ingredients when I was still using the list, but think I accidentally updated it in recipes too. I will update this to keep it by id.

File: client/src/components/SelectedIngredientsList/SelectedIngredientsList.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  setSelectedIngredients,
+}) => {
+  const handleIngredientClick = (ingredientToRemove) => {
+    const updatedIngredients = selectedIngredients.filter(
+      (ing) => ing !== ingredientToRemove
+    );
+    setSelectedIngredients(updatedIngredients);
+  };
+  return (
+    // <div className={styles.wrapper}>
```
Comment (beyzanurck) [inline client/src/components/SelectedIngredientsList/SelectedIngredientsList.jsx:16]: It is important to delete unnecessary comments for the readability of the code.

File: client/src/components/SelectedIngredientsList/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  }
+}
+
+@media (min-width: 1025px) {
+  .selectedContainer {
+    grid-template-columns: repeat(4, 1fr);
+  }
+}
+@media (min-width: 1201px) {
+}
```
Comment (beyzanurck) [inline client/src/components/SelectedIngredientsList/styles.module.scss:41]: This may be deleted as it is never used 😉

**Review by beyzanurck — COMMENTED** [review]
Good job Cathy! It was a pleasure working with you!

File: client/src/components/SelectedIngredientsList/SelectedIngredientsList.jsx (hunk 4 above)
Comment (cathydao95) [reply client/src/components/SelectedIngredientsList/SelectedIngredientsList.jsx:16]: Good catch! Thank you Beyza!

File: client/src/components/SelectedIngredientsList/styles.module.scss (hunk 5 above)
Comment (cathydao95) [reply client/src/components/SelectedIngredientsList/styles.module.scss:41]: Oo! Done :)
